src/resources/2017/Pantogram.ts

```diff
@@ -0,0 +1,224 @@
+import { toInt, visitUrl } from "kolmafia";
+import { have as haveItem } from "../../lib";
+import { $item } from "../../template-string";
+
+const pantogram = $item`portable pantogram`;
+const pants = $item`pantogram pants`;
+
+export function have(): boolean {
+  return haveItem(pantogram);
+}
+
+export function havePants(): boolean {
+  return haveItem(pants);
+}
+
+type PantogramAlignment = "Muscle" | "Moxie" | "Mysticality";
+
+type PantogramElement =
+  | "Hot Resistance: 2"
+  | "Cold Resistance: 2"
+  | "Spooky Resistance: 2"
+  | "Sleaze Resistance: 2"
+  | "Stench Resistance: 2";
+
+type PantogramSacrificeL =
+  | "Maximum HP: 40"
+  | "Maximum MP: 20"
+  | "HP Regen Max: 10"
+  | "HP Regen Max: 15"
+  | "HP Regen Max: 20"
+  | "MP Regen Max: 10"
+  | "MP Regen Max: 15"
+  | "MP Regen Max: 20"
+  | "Mana Cost: -3";
+
+type PantogramSacrificeM =
+  | "Combat Rate: -5"
+  | "Combat Rate: 5"
+  | "Initiative: 50"
+  | "Critical Hit Percent: 10"
+  | "Familiar Weight: 10"
+  | "Candy Drop: 100"
+  | "Item Drop Penalty: -10"
+  | "Fishing Skill: 5"
+  | "Pool Skill: 5"
+  | "Drops Items: true"
+  | "Avatar: Purple";
+
+type PantogramSacrificeR =
+  | "Weapon Damage: 20"
+  | "Spell Damage Percent: 20"
+  | "Meat Drop: 30"
+  | "Meat Drop: 60"
+  | "Item Drop: 15"
+  | "Item Drop: 30"
+  | "Muscle Experience: 3"
+  | "Mysticality Experience: 3"
+  | "Moxie Experience: 3"
+  | "Muscle Experience Percent: 25"
+  | "Mysticality Experience Percent: 25"
+  | "Moxie Experience Percent: 25";
+
```

Comment:
It should be possible to write a bit of TS to extract these from the Alignment, Element tuples, rather than the other way round!

Reply:
yeah, should just be `keyof PantogramAlignment`

Reply:
I don't have a complete image of what you're asking for. Could you please elaborate?